src/lxc/lxccontainer.c

```diff
@@ -5257,6 +5257,18 @@ static int do_lxcapi_seccomp_notify_fd_active(struct lxc_container *c)
 
 WRAP_API(int, lxcapi_seccomp_notify_fd_active)
 
+static bool do_lxcapi_set_timeout(struct lxc_container *c, int timeout)
```

Comment:
Why this generic approach instead of adding a new state api extension that waits with a timeout?